crates/storage/src/connection/block.rs

```diff
@@ -158,26 +158,7 @@ impl Transaction<'_> {
             )
             .context("Deleting block from storage_roots table")?;
 
-        self.inner()
-            .execute(
-                "DELETE FROM trie_class_removals WHERE block_number = ?",
-                params![&block],
-            )
-            .context("Deleting block from trie_class_removals table")?;
-
-        self.inner()
-            .execute(
-                "DELETE FROM trie_contracts_removals WHERE block_number = ?",
-                params![&block],
-            )
-            .context("Deleting block from trie_contracts_removals table")?;
-
-        self.inner()
-            .execute(
-                "DELETE FROM trie_storage_removals WHERE block_number = ?",
-                params![&block],
-            )
-            .context("Deleting block from trie_storage_removals table")?;
+        self.prune_tries(block.into()).context("Pruning tries")?;
```

Comment:
This doesn't quite do what we want here?

There are two issues:
- `prune_tries()` would remove data for `block_number - num_blocks_kept`, which is not what we want here. When purging data of a particular block we'd like to remove data for `block_number`.
- `prune_trie()` removes data for all blocks _before_ `block_number - num_blocks_kept`, not just the data for the exact block we want to purge data for.

Maybe we should stop here and handle the reorg case (the only code path calling `purge_block()`) in a new PR? That needs some more thought...